### PR TITLE
fix: make some improvements

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**
I was working on an example project to get myself familiarize with glee I think these changes will fix a few bugs and make some improvements.
- glee will support multiple messages in channels.
- the glee types will be available for typescript projects.

### CODE REVIEW

File: src/adapters/ws/index.ts [hunk 1]
```diff
@@ -93,15 +93,17 @@ class WebSocketsAdapter extends Adapter {
...
+            
+            ws.on('message', (payload) => {
+              const msg = this._createMessage(pathname, payload)
+              this.emit('message', msg, ws)
+            })
+
+            this.emit('server:connection:open', { name: this.name(), adapter: this, connection: ws, channel: pathname, request })
           })
 
           this.emit('connect', { name: this.name(), adapter: this, connection: ws, channel: pathname })
```
Comment (fmvilas) [inline src/adapters/ws/index.ts:109]: We should get rid of this `connect` event. As per documentation, `connect` is only triggered when Glee is a client but in this case it behaves as a server.

File: src/index.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export default async function GleeAppInitializer () {
     if (channel.hasPublish()) {
       const operationId = channel.publish().json('operationId')
       if (operationId) {
-        const schema = channel.publish().message().payload().json()
+        const schema = {oneOf: channel.publish().messages().map(message => message.payload().json())} as any;
```
Comment (fmvilas) [inline src/index.ts:70]:
```suggestion
        const schema = {oneOf: channel.publish().messages().map(message => message.payload().json())} as any
```

File: src/index.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ export default async function GleeAppInitializer () {
       }
     }
     if (channel.hasSubscribe()) {
-      const schema = channel.subscribe().message().payload().json()
+      const schema = {oneOf: channel.subscribe().messages().map(message => message.payload().json())} as any;
```
Comment (fmvilas) [inline src/index.ts:81]:
```suggestion
      const schema = {oneOf: channel.subscribe().messages().map(message => message.payload().json())} as any
```

File: src/lib/configs.ts [hunk 4]
```diff
@@ -44,9 +44,7 @@ async function loadConfigsFromFile() {
     ASYNCAPI_FILE_PATH = projectConfigs.ASYNCAPI_FILE_PATH ? path.resolve(GLEE_DIR, projectConfigs.ASYNCAPI_FILE_PATH) : ASYNCAPI_FILE_PATH
     return projectConfigs
   } catch (e) {
-    if (e.code !== 'ERR_MODULE_NOT_FOUND') {
       return console.error(e)
```
Comment (fmvilas) [inline src/lib/configs.ts:47]:
```suggestion
    return console.error(e)
```

File: src/lib/lifecycleEvents.ts [hunk 5]
```diff
@@ -79,7 +79,7 @@ export async function run(lifecycleEvent: string, params: GleeFunctionEvent) {
     const responses = await Promise.all(handlers.map(info => info.fn(params)))
     
     responses.forEach(res => {
-      if (res.send) {
+      if (res?.send) {
         res.send.forEach((event: GleeFunctionReturnSend) => {
```
Comment (fmvilas) [inline src/lib/lifecycleEvents.ts:83]: Wouldn't it make more sense to do this directly?

```suggestion
        res?.send?.forEach((event: GleeFunctionReturnSend) => {
```

This way we remove the `if` clause.

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Review by fmvilas — APPROVED** [review]
LGTM 👍 

Pinging @Souvikns @sudoshreyansh @Ruchip16 @atharvagadkari05 too in case they want to review.

**Review by Souvikns — APPROVED** [review]

**Pull request merged**